packages/react-experiments/src/components/SelectedItemsList/Items/subcomponents/DefaultEditingItem.tsx

```diff
@@ -41,118 +45,190 @@ export interface IDefaultEditingItemInnerProps<TItem> extends React.HTMLAttribut
+    const FloatingPicker = props.onRenderFloatingPicker;
     if (!FloatingPicker) {
       return <></>;
     }
     return (
       <FloatingPicker
-        componentRef={this._editingFloatingPicker}
-        onSuggestionSelected={this._onSuggestionSelected}
-        inputElement={this._editingInput}
-        onRemoveSuggestion={this.props.onRemoveItem}
-        onSuggestionsHidden={this.props.onDismiss}
+        componentRef={editingFloatingPicker}
+        onSuggestionSelected={_onSuggestionSelected}
+        targetElement={editingInput.current}
+        onRemoveSuggestion={_onRemoveItem}
+        onFloatingSuggestionsDismiss={props.onDismiss}
+        suggestions={suggestionItems}
+        selectedSuggestionIndex={focusItemIndex}
+        selectedHeaderIndex={headerItemIndex}
+        selectedFooterIndex={footerItemIndex}
+        pickerSuggestionsProps={pickerSuggestionsProps}
       />
     );
   };
 
-  private _resolveInputRef = (ref: HTMLInputElement): void => {
-    this._editingInput = ref;
-
-    this.forceUpdate(() => {
-      this._editingInput.focus();
-    });
-  };
-
-  private _onInputClick = (): void => {
-    this._editingFloatingPicker.current && this._editingFloatingPicker.current.showPicker(true /*updatevalue*/);
-  };
-
-  private _onInputBlur = (ev: React.FocusEvent<HTMLElement>): void => {
-    if (this._editingFloatingPicker.current && ev.relatedTarget !== null) {
-      const target = ev.relatedTarget as HTMLElement;
-      if (
-        target.className.indexOf('ms-Suggestions-itemButton') === -1 &&
-        target.className.indexOf('ms-Suggestions-sectionButton') === -1
-      ) {
-        this._editingFloatingPicker.current.forceResolveSuggestion();
-      }
+  const _onInputBlur = (ev: React.FocusEvent<HTMLElement>): void => {
+    if (focusItemIndex >= 0) {
+      _onSuggestionSelected(ev, suggestionItems[focusItemIndex]);
+    } else if (props.createGenericItem) {
+      onEditingComplete(item, props.createGenericItem(inputValue));
     }
+    // else, we come out of editing mode
   };
 
-  private _onInputChange = (ev: React.FormEvent<HTMLElement>): void => {
+  const _onInputChange = (ev: React.FormEvent<HTMLElement>): void => {
     const value: string = (ev.target as HTMLInputElement).value;
+    setInputValue(value);
 
     if (value === '') {
-      if (this.props.onRemoveItem) {
-        this.props.onRemoveItem(this.props.item);
+      if (props.onRemoveItem) {
+        props.onRemoveItem(item);
       }
     } else {
-      this._editingFloatingPicker.current && this._editingFloatingPicker.current.onQueryStringChanged(value);
+      setEditingSuggestions(getSuggestions(value));
     }
   };
 
-  private _onInputKeyDown(ev: React.KeyboardEvent<HTMLInputElement>): void {
-    if (ev.which === KeyCodes.backspace || ev.which === KeyCodes.del) {
-      ev.stopPropagation();
+  const _onInputKeyDown = (ev: React.KeyboardEvent<HTMLInputElement>): void => {
```

Comment:
I would wrap all these callback functions in `React.useCallback()`.

Reply:
Updated in the new iteration.